### DISCUSSION TITLE
Power Max and Duration

### DISCUSSION THREAD

**Discussion by jms92100** [jmcollin78/solar_optimizer#32] [Q&A]
What is the link between duration and power max ?
I have a water boiler which need 3 hours to get ready.  
it is consuming 3kw/h >>> 9 kw in total   
what should i define  9000 for the power max or 3000 

My swiming pool needs 6 hours and has the need of 1,6kh/h in total this is 9,6kw how do i define the numbers. 

I also have an heatzy what is the best way to define the power and duration.  
any help welcome

**Comment by jmcollin78** [comment]
3kw x 3 h => 9 kwh (energy).
What you need to configure is a power : 3 kw.

**Reply by jms92100** [reply]
Ok thanks for this answer , but could you please add some more info there . 
There are two durations per device, duration_min and duration_stop_min 
I understand this way : 
" When 
   the exported power* selling price* selling tax  >= the power of the device* buying price   
   --> switch on the device  for duration min 
 and when finished 
     stops for duration stop_min."  
But i need to define "the total duration" and what to do if not reached (eq start using the HighCostHP /Low cost HC time) 
for example my swimming pool 1.6kw needs to run for 10hours when the temp is 20°. Is there a way to define those 10 hours with solar optimizer (or should i manage outside solar optimizer) ?
The vmc needs 30 w and could comply with the current definition.
The water boiler is fine also (as it stops when it reaches the defined temp) 

If i can add one more improvment , a priority could be a good way to manage which device to satisfy, the check_usable_template is there to say something like this but can become a nightmare when too many devices to serve 

By the way thank you for this plugin

**Reply by jmcollin78** [reply]
> But i need to define "the total duration" and what to do if not reached (eq start using the HighCostHP /Low cost HC time)
> for example my swimming pool 1.6kw needs to run for 10hours when the temp is 20°. Is there a way to define those 10 hours with solar optimizer (or should i manage outside solar optimizer) ?

Hello,

THis doesn't exists yet but I intend to implement something like this in the future

**Comment by jms92100** [comment]
I have added two sensors for each device . One total needed duration (could be power) and a calculation of the time the switch is on 'or the total power consumed) then i put the condition in the usable template.
for example: check_usable_template: "{{ is_state('sensor.horodateur_ecs', 'on')  and  states('sensor.duree_ecs') | float < states('input_number.ecs_duree_demandee') | float}}"  
and voila.

**Reply by jmcollin78** [reply]
That is a good way to implement the feature. I hope I will have time to implement this natively in the integration.

**Comment by morris2009** [comment]
>I understand this way :
" When
the exported power* selling price* selling tax >= the power of the device* buying price
--> switch on the device for duration min

I presume the above is incorrect and the other way around?
I do have some difficulties getting it all to work so I must have made a mistake somewhere, but it is hard to understand the logic behind it.
If Exporting results in a lower value/kwh then what it would cost to import it, the device needs to be switched on to use the power locally?

Selling Price and buying price are in cents (presumeably)?
Is the tax also in cents, or is that a percentage number?
And the selling price is positive if you receive money for selling a kWh?

In my current setup, it somehow seems I got the basics wrong. I had sun, but the attribute "is usable" = False.
And now the sun is down, that attribute for "is usable" = True.
I do have the the netto_export sensor being negative when power is exported. Did I misunderstood this ?
